examples/speaker_diarization/speaker_diarization/seed_test_suite.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from argparse import ArgumentParser
+from argparse import Namespace
+from typing import Callable
+from typing import Dict
+
+import pandas as pd
+from tqdm import tqdm
+from workflow import GroundTruth
+from workflow import TestCase
+from workflow import TestSample
+from workflow import TestSuite
+
+import kolena
+from kolena.workflow.annotation import LabeledTimeSegment
+
+BUCKET = "kolena-public-datasets"
+DATASET = "ICSI-corpus"
+
+
+def seed_test_suite_by_avg_amp(
+    test_suite_name: str,
+    complete_test_case: TestCase,
+) -> TestSuite:
+    test_case_name_to_decision_logic_map = {
+        "1st tertile": lambda x: x < 0.0549,
+        "2nd tertile": lambda x: 0.0549 <= x < 0.102,
+        "3rd tertile": lambda x: 0.102 <= x,
+    }
```

Comment:
Would be nice to automatically compute these based on the contents of the test samples — seems it shouldn't be too challenging to refactor this function to load test samples upfront, partition into tertiles, then iterate through to create test cases.